tests/Unit/Security/Authenticator/JwtTokenAuthenticatorTest.php

```diff
@@ -0,0 +1,113 @@
+{
+    public function testItDoesNotSupportRememberMe(): void
+    {
+        $subject = new JwtTokenAuthenticator(
+            $this->createMock(AuthorizationHeaderTokenExtractor::class),
+            $this->createMock(JwtTokenVerifier::class),
+            $this->createMock(SerializerResponder::class)
+        );
+
+        self::assertFalse($subject->supportsRememberMe());
+    }
+
+
+    public function testItExtractsCredentialsCorrectly(): void
+    {
+        $subject = new JwtTokenAuthenticator(
+            $this->createMock(AuthorizationHeaderTokenExtractor::class),
+            $this->createMock(JwtTokenVerifier::class),
+            $this->createMock(SerializerResponder::class)
+        );
+
+        $extractor = new AuthorizationHeaderTokenExtractor();
+
+        $request = $this->getMockBuilder(Request::class)
+            ->getMock();
+        ;
+
+        $headerBag = $this->getMockBuilder(HeaderBag::class)
+            ->setConstructorArgs([
+                [
+                    'Authorization' => 'Bearer expected'
+                ]
+            ])
+            ->onlyMethods(['has', 'get'])
+            ->getMock()
+        ;
+        $headerBag->method('has')->with('Authorization')->willReturn(true);
+        $headerBag->method('get')->with('Authorization')->willReturn('Bearer expected');
+
+        $request->headers = $headerBag;
+
+        $subject->setExtractor($extractor);
+
+        self::assertSame('expected', $subject->getCredentials($request));
+    }
+
+    public function testItThrowsExceptionOnNoUsernameClaimInToken(): void
+    {
+        $subject = new JwtTokenAuthenticator(
+            $this->createMock(AuthorizationHeaderTokenExtractor::class),
+            $this->createMock(JwtTokenVerifier::class),
+            $this->createMock(SerializerResponder::class)
+        );
+
+        $tokenMock = $this->getMockBuilder(Token::class)
+            ->getMock();
+
+        $tokenMock->expects($this->any())->method('verify')->willReturn(false);
+
+        $userProviderMock = $this->getMockBuilder(UserProviderInterface::class)
+            ->getMock();
+
+        $this->expectException(AuthenticationException::class);
+        $this->expectExceptionMessage('Invalid token. Unable to parse or no username claim.');
+
+        // @codingStandardsIgnoreStart
+        $subject->getUser(
+            'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwidXNlciI6InVzZXIxIiwiaWF0IjoxNTE2MjM5MDIyfQ.MqMn8PLjkMH_0pAmkVXg6FolaiaKyZZ_Bqnt-xS50CM',
+            $userProviderMock
+        );
+        // @codingStandardsIgnoreEnd
+    }
+
+    public function testItThrowsExceptionOnInvalidToken(): void
+    {
+        $subject = new JwtTokenAuthenticator(
+            $this->createMock(AuthorizationHeaderTokenExtractor::class),
+            $this->createMock(JwtTokenVerifier::class),
+            $this->createMock(SerializerResponder::class)
+        );
+
+        $tokenMock = $this->getMockBuilder(Token::class)
```

Comment:
Suggestion: use $this->createMock instead

Reply:
Because I do not believe that you need the original constructor, clone methods in this case.